### ISSUE TITLE
(Potential) Flutter lint: "use containers"

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#2059]
There are some cases where a `Container` is preferred.

@jacob314 you made a comment in https://github.com/flutter/devtools/pull/1764/files/7c8a06e3ca03ca7f657e4b1cbbf209a5200b1793#r398140772

on this code:

```dart
        Padding(
          padding: DevToolsScaffold.appPadding,
          child: Align(
            alignment: Alignment.topLeft,
            child: Column(
              children: <Widget>[
                BannerMessageContainer(screen: screen),
                Expanded(
                  child: screen.build(context),
                ),
              ],
            ),
```

Could you elaborate?  What are conditions that indicate a `Container` should be introduced?

**Comment by jacob314** [comment]
Any time using a container would reduce the number of widgets required it should be used.
That reduces the level of nesting and makes it clearer what is going on.
 For example, in that case using container lets you go from an Align + Padding widget to just a Container.  If the Align was missing then using Padding instead of Container is just fine.

**Comment by pq** [comment]
Awesome.  Thanks for the follow-up @jacob314! 👍

**Comment by a14n** [comment]
One restriction through. Container can't ne const.

**Comment by jacob314** [comment]
Fyi @munificent. Will the new proposals for enhanced const allow us to make Container const? It would be nice if using Container for terser code didn't degrade performance.

**Comment by munificent** [comment]
"Will" -> "would", since it's definitely not clear whether Enhanced Const is feasible yet. But, yes, if we had that I believe we could make the Container constructor const.
